specification/storage/resource-manager/Microsoft.Storage/preview/2020-08-01-preview/storage.json

```diff
@@ -2160,6 +2170,11 @@
           "$ref": "#/definitions/KeyVaultProperties",
           "x-ms-client-name": "KeyVaultProperties",
           "description": "Properties provided by key vault."
+        },
+        "identity": {
+          "$ref": "#/definitions/EncryptionIdentity",
+          "x-ms-client-name": "Identity",
+          "description": "Encryption Identity properties when storage account is associated with a UserAssignedIdentity."
```

Comment:
Could user both assign a user identity here, and also assign a system identity?

Reply:
No. "EncryptionIdentity" property is only for UserAssignedIdentities.

Reply:
Then if user try to assign system identity to an storage account which already has user assigned identity, what will happen? vise version?
It seems we didn't block user to do that from rest API definition.

Reply:
I think the 'Identity' property you are referring to is the property on the storage account level which can be either 'SystemAssigned' and 'UserAssigned'. The changes in the current PR is not modifying that property.

The "EncryptionIdentity" here is referring to a new 'identity' property within the scope of the EncryptionProperties of the storage account. These properties only apply to UserAssignedIdentities.

Reply:
Thanks for the reply!
Is there any relationship of the 2 identities setting?
Is there any feature spec rest API spec for this change? After user set this identity, what will happen?

Reply:
Hi Sorry for the late reply. I got pulled into higher priority tasks for two weeks.

I will confirm the details with feature owner and reply here soon

Reply:
EncryptionIdentity allows you to specify the specific user assigned identity that needs to be used for encryption. When not applied, Storage defaults to attempting to use the SystemAssigned Identity.